src/aap_eda/api/views/activation.py

```diff
@@ -54,14 +54,16 @@ class ActivationViewSet(
     mixins.DestroyModelMixin,
     viewsets.GenericViewSet,
 ):
-    queryset = models.Activation.objects.all()
     serializer_class = serializers.ActivationSerializer
     filter_backends = (defaultfilters.DjangoFilterBackend,)
     filterset_class = filters.ActivationFilter
 
     rbac_resource_type = None
     rbac_action = None
 
+    def get_queryset(self):
+        return models.Activation.access_qs(self.request.user)
+
```

Comment:
we should replace line 111 in this file to call this since we are writing a custom list endpoint, otherwise I'm able to view activations with a user that has no roles for activation.

Reply:
same in retrieve endpoint.

Reply:
In similar cases I've seen a different solution to this problem, like this:

```python
    def get_serializer_class(self):
        if self.action == 'update':
            return RoleDefinitionDetailSeraizler
        return super().get_serializer_class()
```

Would that general method be okay to use here? That would involve some changes to the list/retrieve overrides.

Reply:
The solution I was thinking was replacing `activations = models.Activation.objects.all()` with `activations = self.get_queryset()` in the following line. That would handle access to data on model level.
https://github.com/AlanCoding/eda-server/blob/a1156374b3c4b519206086b838753f2d6defb61a/src/aap_eda/api/views/activation.py#L111